packages/atomic/src/components/result-template-components/atomic-result-value/atomic-result-value.tsx

```diff
@@ -12,17 +12,38 @@ import {
 export class AtomicResultValue {
   @Element() host!: HTMLDivElement;
   @Prop() value = '';
+  @Prop() shouldHighlightWith = '';
 
   @ResultContext() private result!: Result;
 
+  /**
+   * Returns the value of 'shouldHighlightWith' prop from the Result
+   */
+  private getHighlightsFromResult() {
+    const highlightsKeywords = Object.entries(this.result).find(
+      (keyValue) => keyValue[0] === this.shouldHighlightWith
+    );
+    return highlightsKeywords![1] as string[];
+  }
+
   @ResultContextRenderer
-  public render() {
-    const resultValue = ResultTemplatesHelpers.getResultProperty(
+  render() {
+    let resultValue = ResultTemplatesHelpers.getResultProperty(
       this.result,
       this.value
     );
+
     if (resultValue !== null) {
-      return resultValue;
+      if (this.shouldHighlightWith && this.shouldHighlightWith in this.result) {
```

Comment:
I'm not clear on the second half of this condition. It is checking that `this.shouldHighlightWith` (which is a non-empty string) is a key of the result object?

Reply:
I update the PR and removed the second condition. I change the type on `shouldHighlightWith` so it's guaranteed to be in the result object.